libcaf_net/caf/net/web_socket/framing.cpp

```diff
@@ -267,9 +267,11 @@ class framing_impl : public framing {
     if (hdr_.opcode == detail::rfc6455::text_frame
```

Comment:
Here, we need that `if constexpr (sizeof(size_t) < sizeof(uint64_t))` logic to validate `hdr_.payload_len` before we cast it. But not with an assertion. If the `payload_len` exceeds what we can represent on this platform, we need to call `abort_and_shutdown` with a descriptive error.